### ISSUE TITLE
It's not possible to use labels in custom rules

### ISSUE THREAD

**Issue by alberpilot** [wazuh/wazuh#3130]
|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| All | Manager | Packages/Sources | All |

Hello team

  After testing Wazuh, we noticed that it's not possible to use labels inside rules. Below, you'll find a use case. 

We define a label in the desired agent. Example: 
```
[root@manager vagrant]# cat /var/ossec/etc/shared/default/agent.conf 
<agent_config>

  <labels>
    <label key="agentgroups">docker</label>
  </labels>

</agent_config>
```
This configuration will be pushed to the agent in the default group. 

In the manager side, we have the following custom rule: 
```
  <rule id="100023" level="12">
    <if_sid>5402</if_sid>
    <field name="agent.labels.agentgroups">docker</field> 
    <description>Custom children rule!!!!!</description>
  </rule>
```
But rule 100023 will never appear. It's not possible to use the label value, agent.ip, agent.id either. 

We consider very usefully if we could use all the alert fields in rules.  

Best regards, 
Alberto R

**Comment by branchnetconsulting** [comment]
Please make this work.  It would massively increase the value of agent labels to let them be used in Wazuh rule criteria.  I just had a student asking about a use case that really needs this feature to work to solve their issue.

**Comment by branchnetconsulting** [comment]
In addition to agent labels, I propose that syscheck tags also be visible to Wazuh rules.  I imagine it would be easiest to address both of these at the same time.

In fact, I would strongly suggest that by design we always make new Wazuh field enrichments visible to Wazuh rules.  Like I mentioned in #3415, enrichment not only feeds richer data to Kibana/Splunk, but could be fueling better rule-based analysis of our events.

Kevin

**Comment by jchenrev** [comment]
Hello there, any updates of this feature? Is it going to be implemented in 4.x?
It would be very useful for me as well. Because I labelled agents by environments, roles. 
Some environments can be more important than other environments so I want to adjust level by environment.
Thank you!

**Comment by devinacosta** [comment]
Would love to see this feature. :) +1

**Comment by Crankinator** [comment]
Wanted to bump this and see if there were any updates here, or if they will be implemented in 4.x.
I feel like this, with the addition of Kevin's (branchnetconsulting) #3415 issue, this will allow granularity into specific groups that need to be monitored at a more specific level. This will be _EXTREMELY_ useful for complex infrastructures.

**Comment by stiliyana-simeonova** [comment]
Bumping this up, 
Came across the issue after crafting a rule based on agent labels with no luck.
It would be very very useful to have this feature

**Comment by mtridici** [comment]
Would like to see this feature. It will be very useful. +1

**Comment by kmfreder1** [comment]
Is there any update on this?  The suggestions by @alberpilot and @branchnetconsulting would be so helpful in multiple customer contexts.

**Comment by mimugmail** [comment]
I also like this and have a discussion here:
https://wazuh.slack.com/archives/C0A933R8E/p1626322640174200

**Comment by branchnetconsulting** [comment]
I see above this was closed as not planned and also moved to Global development as done.  Does this mean that robust support for agent labels and syscheck tags being used as criteria in rules is planned for the 5.x engine, or not planned at all?

**Comment by ezequielleiter** [comment]
Please make this work, it's a very helpful tool.

**Comment by alexgot1151** [comment]
Hi team,

Is there any plan to introduce functionality for applying rules to specific agent groups in Wazuh? It would be extremely beneficial to have the ability to filter rules based on agent groups or labels—something like group-based rule scoping.

Currently, Wazuh rules don't seem to be group-aware, so managing rule applicability across different environments (e.g., dev, prod, test) becomes difficult without manually customizing rules based on hostnames, labels, or CDB lists. While using labels and hostname conditions offers a workaround, a native way to scope rules to agent groups directly would be much more manageable and scalable.

Is this feature planned for a future release, or is there an existing roadmap item related to this? Would love to hear your thoughts or guidance on best practices for this kind of group-based rule management.

Thanks in advance!

**Comment by branchnetconsulting** [comment]
I also really hope that rule level group awareness is part of Wazuh 5.x.
For now, hostname or CDB hostname list criteria is the closest we can get,
since Wazuh rules cannot use agent labels as matching criteria.  In fact it
would be a cool project to automatically sync CDBs to the membership of
select agent groups that would be useful to use as rule matching criteria.
Here are some other possibilities we could pursue while waiting for Wazuh
to fundamentally resolve this limitation:

   - Extend the ingest node pipeline to drop or somehow escalate certain
   alerts involving a given rule.id based on the content of the rule.groups
   field in the alert record.
   - Build custom active responses and/or integrations that abort if the
   alert passed to them has a specific group in the rule.groups field that
   should be excluded.
   - Use query based detection for agent-group-specific scenarios,  via the
   OpenSearch Alerting and Notifications plugins.
   (
   https://wazuh.com/blog/exploring-security-alerting-options-for-improved-threat-detection-in-wazuh-part-1/
   )

**Comment by dathannobrega** [comment]
Hello Guys, any notice about this feature?
 It would be extremely beneficial to have the ability to filter rules based on agent groups or labels

**Comment by CauXZO** [comment]
Any news about it?

**Comment by branchnetconsulting** [comment]
> Any news about it?

I suspect that we will find this has been directly or indirectly resolved in Wazuh 5.0.  I hear that is soon to hit public beta.

**Comment by CauXZO** [comment]
Wow, Nice!
Ty for the asnwer, friend!
